staging/src/k8s.io/component-base/logs/logrotation/logrotation.go

```diff
@@ -0,0 +1,233 @@
+/*
+Copyright 2024 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package logrotation
+
+import (
+	"io"
+	"os"
+	"path/filepath"
+	"strings"
+	"sync"
+	"time"
+)
+
+const timeLayout string = "20060102-150405"
+
+type RotationFile struct {
```

Comment:
Let's not add this as public API of `k8s.io/component-base/logs`.

Can you move the entire package to `k8s.io/component-base/logs/kube-log-runner/internal/logrotation`?

Also, let's be very clear about what the public API of that package is. Only things that are needed from outside should get exported. This `RotationFile` should be `rotationFile`.

Reply:
yes, I can move. just thought that it might be useful for tools other than the kube-log-runner as well